django/db/models/fields/related_objects.py

```diff
@@ -0,0 +1,315 @@
+"""
+"Rel objects" for related fields.
+
+"Rel objects" (for lack of a better name) carry information about the relation
+modeled by a related field and provide some utility functions. They're stored
+in the ``rel`` attribute of the field.
```

Comment:
Maybe we could be bold here:
1. Rename `related_objects.py` into `related_fields.py`.
2. Move all fields from `related.py` to `related_fields.py`.
3. Rename all `*Rel` objects into `Reverse*Field`, then later we'll figure out how to actually implement them as fields.

As far as the Meta API, these are now fields already.